### ISSUE TITLE
Open URL in New Window ... video re-buffers when you put it in a File Loop

### ISSUE THREAD

**Issue by sxflynn** [iina/iina#3945]
Steps to reproduce:

1. File -> Open URL in New Window...
2. Put in a Youtube URL so it's uses youtube-dl
3. Select Playback --> File Loop
4. When the video is finished playing, it starts at the beginning again but has to rebuffer.

The expected behavior in Step 4 is that the video starts over again but doesn't have to rebuffer or redownload because the video has already been stored in memory/RAM somehow.

Is there a way IINA could be improved so that when File Loop is selected, it just buffers the whole video locally and then when it's time to loop the video it just immediately replays the video without having to buffer all over again?

**Comment by anohren** [comment]
I agree that it should work like that in theory. Videos can be arbitrarily large though, so if it works the way you describe then there's probably only one general behavior in mpv for all sizes. My guess is that the quickest path forward would be if you could convince mpv that they should cache videos where the video is estimated to be smaller than some size (which reasonably varies between machines). Or maybe that's already supported and just IINA doesn't make use of it yet...

**Comment by sxflynn** [comment]
Perhaps the solution is for IINA to detect if it's less than 10 seconds from the end of the video AND file loop is enabled AND the open video is an open URL buffer. If those conditions are all true, it will begin pre-buffering the beginning of the video so right at T minus 0 it can immediately begin playing the beginning of the video since it began buffering in the background.
